src/go/rpk/pkg/config/params.go

```diff
@@ -552,3 +568,133 @@ func (c *Config) addUnsetDefaults() {
 		r.AdminAPI.Addresses = []string{"127.0.0.1:9644"}
 	}
 }
+
+// Set allow to set a single configuration property by passing a key value pair
+//
+//   Key:    string containing the yaml property tag, e.g: 'rpk.admin_api'.
+//   Value:  string representation of the value, either single value or partial
+//           representation if a format is passed (yaml / json).
+//   Format: either single, json, or yaml (default: single).
+func (c *Config) Set(key, value, format string) error {
+	if key == "" {
+		return fmt.Errorf("key field must not be empty")
+	}
+	props := strings.Split(key, ".")
+	rv := reflect.ValueOf(c).Elem()
+	found, other, err := getField(props, rv)
+	if err != nil {
+		return err
+	}
+	isOther := other != reflect.Value{}
+
+	field := found
+	if isOther {
+		field = other
+	}
+
+	if field.CanAddr() {
+		i := field.Addr().Interface()
+		in := value
+		switch strings.ToLower(format) {
+		case "yaml", "single", "":
```

Comment:
Seeing how `single` is the same as `yaml`, can we remove every other mention of `single` in set flags? I've never understood it -- single _is_ yaml, and yaml is clearer.

Reply:
Some users might have `--single` flag in their scripts. wdyt of hiding the flag and/or single to be an alias for yaml?

Reply:
:+1: to the idea of removing all outward facing references of `single` but keeping it for backcompat

Reply:
Since we improved the help text I'll just leave: 

`--format      Format of the value (yaml/json)`